dev-client/src/screens/LocationScreens/components/soilInfo/SoilInfoDisplay.tsx

```diff
@@ -73,22 +74,69 @@ type SoilInfoDisplayUSProps = {
   soilInfo: SoilInfo;
 };
 
+function getNormalizedSoilName(soilSeriesName: string) {
+  let normalizedSoilName = soilSeriesName
+    .trim()
+    .toLowerCase()
+    .replace(/ /g, '_');
+
+  // TODO: Fix this typo in the database instead of handling it here
+  if (normalizedSoilName === 'albic_luvsiols') {
+    normalizedSoilName = 'albic_luvisols';
+  }
+
+  return normalizedSoilName;
+}
+
+function getSoilSeriesDisplayText(
+  soilSeries: SoilSeries,
+  t: TFunction,
+  i18n: i18nType,
+) {
+  // As of 2025-07 we only expect global (not US) soil match descriptions to come from the client-side i18n files
+  const soilKey = `soil.match_info.${getNormalizedSoilName(soilSeries.name)}`;
+
+  if (i18n.exists(soilKey)) {
+    const soilSeriesTextForDisplay: SoilSeries = {
+      name: i18n.exists(`${soilKey}.name`)
+        ? t(`${soilKey}.name`)
+        : soilSeries.name,
+      description: i18n.exists(`${soilKey}.description`)
+        ? t(`${soilKey}.description`)
+        : soilSeries.description,
+      management: i18n.exists(`${soilKey}.management`)
+        ? t(`${soilKey}.management`)
+        : soilSeries.management,
+      fullDescriptionUrl: soilSeries.fullDescriptionUrl ?? undefined,
+      taxonomySubgroup: soilSeries.taxonomySubgroup ?? undefined,
+    };
+    return soilSeriesTextForDisplay;
+  } else {
+    return soilSeries;
+  }
+}
+
 function SoilInfoDisplayGlobal({soilInfo}: SoilInfoDisplayGlobalProps) {
-  const {t} = useTranslation();
+  const {t, i18n} = useTranslation();
+  const displaySoilSeries = getSoilSeriesDisplayText(
+    soilInfo.soilSeries,
+    t,
+    i18n,
+  );
 
   return (
     <Column space="0px">
-      {soilInfo.soilSeries.description && (
+      {displaySoilSeries?.description && (
```

Comment:
nit: following the code, it seems like `displaySoilSeries` should never need a `.?`?